src/main/kotlin/no/nav/familie/ks/sak/barnehagelister/BarnehagelisteVarslingService.kt

```diff
@@ -0,0 +1,94 @@
+
+import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import no.nav.familie.ks.sak.barnehagelister.BarnehagelisteVarslingService.Companion.PATH_ENHETSNR_TIL_KOMMUNENR_OVERSIKT
+import no.nav.familie.ks.sak.barnehagelister.domene.BarnehagebarnRepository
+import no.nav.familie.ks.sak.barnehagelister.epost.EpostService
+import no.nav.familie.ks.sak.common.exception.Feil
+import org.slf4j.LoggerFactory
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.io.File
+import java.time.LocalDateTime
+
+@Service
+class BarnehagelisteVarslingService(
+    val barnehageBarnRepository: BarnehagebarnRepository,
+    val epostService: EpostService,
+) {
+    @Scheduled(cron = "0 0 6 * * ?")
+    @Transactional
+    fun sendVarslingOmNyBarnehagelisteTilEnhet() {
+        val kommunerSendtForFørsteGangSisteDøgn = finnKommunerSendtInnSisteDøgn()
+        val enheterSomSkalVarslesTilKommuner = kommunerSendtForFørsteGangSisteDøgn.groupBy { finnEnhetForKommune(it) }
+
+        if (enheterSomSkalVarslesTilKommuner.isNotEmpty()) {
+            logger.info("Sender epost for nye kommuner i barnehagelister: $kommunerSendtForFørsteGangSisteDøgn")
+
+            enheterSomSkalVarslesTilKommuner
+                .map { (enhetsNr, kommuneNrTilhørendeEnhet) -> finnKontaktEpostForEnhet(enhetsNr) to kommuneNrTilhørendeEnhet.map { finnKommuneNavnForKnr(it) }.toSet() }
+                .forEach { (epostAdresse, kommunerTilhørendeEnhet) ->
+                    epostService.sendEpostVarslingBarnehagelister(epostAdresse, kommunerTilhørendeEnhet)
+                }
+        }
+    }
+
+    // Bruker kommuneNr i stedet for kommunenavn siden det er mindre mulighet for feilskriving
+    @Transactional
+    fun finnKommunerSendtInnSisteDøgn(): Set<String> {
+        val barnSendtInnSisteDøgn =
+            barnehageBarnRepository
+                .findAll()
+                .filter { it.endretTidspunkt >= LocalDateTime.now().minusDays(1) }
+        return barnSendtInnSisteDøgn.map { it.kommuneNr }.toSet()
+    }
+
+    private fun finnKontaktEpostForEnhet(enhetsNr: String): String =
+        when (enhetsNr) {
+            "4820" -> `KONTAKT_E-POST_VADSØ`
+            "4812" -> `KONTAKT_E-POST_BERGEN`
+            else -> {
+                throw Feil("Det er ikke lagret noen epost for å kontakte enhet $enhetsNr.")
+            }
+        }
+
+    companion object {
+        val `KONTAKT_E-POST_VADSØ` = "nav.familie-.og.pensjonsytelser.vadso.kontantstotte@nav.no"
+        val `KONTAKT_E-POST_BERGEN` = "kontantstotte@nav.no"
+
+        val PATH_ENHETSNR_TIL_KOMMUNENR_OVERSIKT =
+            "src/main/resources/barnehagelister/barnehagelister-enhet-til-kommune.json"
+
+        private val logger = LoggerFactory.getLogger(BarnehagelisteVarslingService::class.java)
+    }
+}
+
+data class Kommune(
```

Comment:
Denne kan kanskje være `private`? 

```suggestion
private data class Kommune(
```